invenio_users_resources/records/api.py

```diff
@@ -8,62 +8,111 @@
 # details.
 
 """API classes for user and group management in Invenio."""
-import random
+
 import unicodedata
+from collections import namedtuple
+from datetime import datetime
 
 from flask import current_app
+from invenio_accounts.models import Domain
 from invenio_accounts.proxies import current_datastore
 from invenio_db import db
-from invenio_records.dumpers import SearchDumper
-from invenio_records.systemfields import DictField, ModelField
+from invenio_records.dumpers import SearchDumper, SearchDumperExt
+from invenio_records.dumpers.indexedat import IndexedAtDumperExt
+from invenio_records.systemfields import ModelField
 from invenio_records_resources.records.api import Record
 from invenio_records_resources.records.systemfields import IndexField
+from sqlalchemy.exc import NoResultFound
 
 from .dumpers import EmailFieldDumperExt
-from .models import GroupAggregateModel, UserAggregateModel
-
-
-def parse_user_data(user):
-    """Parse the user's information into a dictionary."""
-    data = {
-        "id": user.id,
-        "email": user.email,
-        "username": user.username,
-        "active": user.active,
-        "confirmed": user.confirmed_at is not None,
-        "preferences": dict(user.preferences or {}),
-        "profile": dict(user.user_profile or {}),
-        "blocked_at": user.blocked_at,
-        "verified_at": user.verified_at,
-    }
-
-    data["preferences"].setdefault("visibility", "restricted")
-    data["preferences"].setdefault("email_visibility", "restricted")
-    default_locale = current_app.config.get("BABEL_DEFAULT_LOCALE", "en")
-    data["preferences"].setdefault("locale", default_locale)
-    data["preferences"].setdefault("timezone", "Europe/Zurich")
-    data["preferences"].setdefault(
-        "notifications",
-        {
-            "enabled": True,
-        },
-    )
+from .models import DomainAggregateModel, GroupAggregateModel, UserAggregateModel
+from .systemfields import (
+    AccountStatusField,
+    AccountVisibilityField,
+    DomainCategoryNameField,
+    DomainField,
+    DomainOrgField,
+    DomainStatusNameField,
+    IsNotNoneField,
+    UserIdentitiesField,
+)
+
+EmulatedPID = namedtuple("EmulatedPID", ["pid_value"])
+"""Emulated PID"""
+
+
+class AggregatePID:
```

Comment:
Provides an interface that is needed by Records-Resources service layer (otherwise we have to implement all the service methods).